libp2p/network/swarm.py

```diff
@@ -270,11 +270,9 @@ def remove_conn(self, swarm_conn: SwarmConn) -> None:
         """Simply remove the connection from Swarm's records, without closing
         the connection."""
         peer_id = swarm_conn.muxed_conn.peer_id
-        if peer_id not in self.connections:
-            return
         # TODO: Should be changed to remove the exact connection,
         #   if we have several connections per peer in the future.
-        del self.connections[peer_id]
+        self.connections.pop(peer_id, None)
```

Comment:
I would advocate to **only** use this in situations where you explicitely want graceful failure handling.  Otherwise this might hide bugs.  Fox example, this previously had no error handling around it so now this would silently pass when previously it would have raised an error which might be a good thing depending on the code path.